kwok/charts/crds/karpenter.sh_nodeclaims.yaml

```diff
@@ -361,6 +361,9 @@ spec:
                 - nodeClassRef
                 - requirements
               type: object
+              x-kubernetes-validations:
+                - message: immutable field changed
+                  rule: self == oldSelf
```

Comment:
Are we saying this doesn't apply to metadata? I'm seeing the title of the PR say spec, so I'm wondering what the decision is. I'm thinking it has to be mutable, since we have to add these labels in ourselves at nodeclaim launch

Reply:
Yes, we currently want to make sure the metadata is mutable. The goal here is to only make node claim spec immutable. v1 RFC: https://github.com/kubernetes-sigs/karpenter/pull/1222/files#diff-5721182d9716c35869521e76985d0593716478602bf2a8e4862303ef8f3d5f06R310

Reply:
Since, the Karpenter controller makes updates to the annotation and labels after node claim creation we need to make sure the metadata object is mutable.